File: backend/src/main/java/dev/visie/elections/model/election/Candidate.java
package dev.visie.elections.model.election;


import com.fasterxml.jackson.annotation.JsonBackReference;
import dev.visie.elections.model.election.compositeId.CandidateId;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
@Entity
public class Candidate {

    @EmbeddedId
    private CandidateId candidateId;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name_prefix")
    private String lastNamePrefix;

    @Column(name = "last_name")
    private String lastName;
    private String initails;
    private String locality;


    @ManyToOne
    @JoinColumn(name = "party_id", referencedColumnName = "party_id", insertable = false, updatable = false)
    @JsonBackReference
    private Party party;

    @Override
    public String toString() {
        return "Candidate{" +
                "id=" + candidateId +
                ", firstName='" + firstName + '\'' +
                ", lastNamePrefix='" + lastNamePrefix + '\'' +
                ", lastName='" + lastName + '\'' +
                ", location='" + locality + '\'' +
                ", partyId=" + (party != null ? party.getPartyId() : "null") +
                '}';
    }
}

File: frontend/src/models/Candidate.ts
export interface Candidate {
  candidateId: number;
  firstName: string;
  lastName: string;
  initials: string;
  lastNamePrefix: string;
  locality: string;
  partyId: number;
}

export interface CandidateWithVotes {
  candidateId: CandidateId
  fullName: string;
  votes: number;
}

type CandidateId = {
  candidateId: number;
  partyId: number;
}